tests/figures/figure_component.test.ts

```diff
@@ -330,106 +382,130 @@ describe("figures", () => {
+        expect(model.getters.getFigure(sheetId, id)?.anchor).toMatchObject({
+          col: 16,
+          row: 4, // initial position - scroll offset
         });
       });
 
       test("Figure in frozen cols can be dragged to main viewport", async () => {
-        createFigure(model, { id, x: 4 * cellWidth, y: 16 * cellHeight });
+        createFigure(model, { id, anchor: { col: 4, row: 16 } });
         await nextTick();
         await dragElement(figureSelector, { x: 3 * cellWidth, y: 0 }, undefined, true);
-        expect(model.getters.getFigure(sheetId, id)).toMatchObject({
-          x: 17 * cellWidth, // initial position + drag offset + scroll offset
-          y: 16 * cellHeight,
+        expect(model.getters.getFigure(sheetId, id)?.anchor).toMatchObject({
+          col: 17, // initial position + drag offset + scroll offset
+          row: 16,
         });
       });
 
       test("Figure in main viewport can be dragged to frozen cols", async () => {
-        createFigure(model, { id, x: 16 * cellWidth, y: 16 * cellHeight });
+        createFigure(model, { id, anchor: { col: 16, row: 16 } });
         await nextTick();
         await dragElement(figureSelector, { x: -3 * cellWidth, y: 0 }, undefined, true);
-        expect(model.getters.getFigure(sheetId, id)).toMatchObject({
-          x: 3 * cellWidth, // initial position + drag offset - scroll offset
-          y: 16 * cellHeight,
+        expect(model.getters.getFigure(sheetId, id)?.anchor).toMatchObject({
+          col: 3, // initial position + drag offset - scroll offset
+          row: 16,
         });
       });
 
       test("Dragging figure that is half hidden by frozen cols will put in on top of the freeze pane", async () => {
-        createFigure(model, { id, x: 14 * cellWidth, y: 16 * cellHeight, width: 5 * cellWidth });
+        createFigure(model, {
+          id,
+          anchor: { col: 14, row: 16 },
+          width: 5 * cellWidth,
+        });
         await nextTick();
         await dragElement(figureSelector, { x: 0, y: 1 }, undefined, true);
-        expect(model.getters.getFigure(sheetId, id)).toMatchObject({
-          x: 4 * cellWidth, // initial position - scroll offset
-          y: 16 * cellHeight + 1,
+        expect(model.getters.getFigure(sheetId, id)?.anchor).toMatchObject({
+          col: 4, // initial position - scroll offset
+          row: 16,
         });
       });
     });
 
     test.each([
-      [{ wheelX: 0, wheelY: 10 * DEFAULT_CELL_HEIGHT }],
-      [{ wheelX: 10 * DEFAULT_CELL_WIDTH, wheelY: 0 }],
-      [{ wheelX: 0, wheelY: 50 * DEFAULT_CELL_HEIGHT }], // scroll out of original viewport
-      [{ wheelX: 40 * DEFAULT_CELL_WIDTH, wheelY: 0 }], // scroll out of original viewport
+      [{ wheelX: 0, wheelY: 10 }],
+      [{ wheelX: 10, wheelY: 0 }],
+      [{ wheelX: 0, wheelY: 50 }], // scroll out of original viewport
+      [{ wheelX: 40, wheelY: 0 }], // scroll out of original viewport
```

Comment:
nitpick: variable names `wheelX` which looks like DOM event parameters being in `HeaderOffset` instead of `Pixel` is a bit confusing. Just add `additionalOffsetX` or keep wheelX and do things like `Math.floor(wheelX /DEFAULT_CELL_HEIGHT)` to compute the anchor/offset if you want to test those